### DISCUSSION TITLE
Save / Load of chats

### DISCUSSION THREAD

**Discussion by abhuva** [brianpetro/obsidian-smart-connections#86] [Ideas]
I see that there are json files appearing that sometimes hold parts of the chat.
But atleast for now, this seems inconsistent - its not clear how to trigger the save.

I would like to have the possibility to save a conversation and also later reload a conversation (pretty much the same chatgpt is doing it). Beside its being convenient to actually look at older conversations, it can also save a lot of tokens in the long run - when i dont have to ask a certain question again.

I could imagine having a simple Save / Load button on the top of the chat window (where the + button is wich appearantly opens a new chat).

**Comment by brianpetro** [comment]
Hi @abhuva ,

Good noticing the files. The user interface for accessing the chat history is coming very soon!

To clarify why it seems like only some chats are saved, that's because, currently, the only save trigger is when you use the "+" button to create a new chat. However, saving will become automatic once the feature is officially released.

Thanks for joining the discussion,
Brian🌴

**Reply by abhuva** [reply]
Thanks for the fast answer - its good to know this is on the roadmap, no problem with beeing a little bit patient =)

**Reply by brianpetro** [reply]
@abhuva `v1.4.3` adds a button to make the history accessible.

Any history saved before this update is likely unusable.

The only additional "save" trigger is if you close the chat window. The "Save" button will be coming soon.

**Comment by pinuke** [comment]
Saving chats was proposed with #59. I don't believe there is a Feature Request (FR) yet for loading chat history. You're welcome to open up a new issue thread and make one.

**Comment by brianpetro** [comment]
Chat history UI added in `v1.4.6`

**Reply by renanpoco** [reply]
Hi, first of all thanks for this amazing plugin, i am learning how to use obsidian and i just can't find where is the chatgpt history inside obsidian. Can you please help me ? Thanks
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/17159282/c941b402-e92c-42bb-8208-8859e6a2dd51)

**Reply by brianpetro** [reply]
You can view past chats by clicking the button on the top right of the chat window and the chats file is in the `.smart-connections` folder in the root of your vault.

🌴

**Comment by frankcsliu** [comment]
I found no problem starting a chat with a title and hitting the safe bottom. But it shows an "invalid chat_id" error message every time I hit the save bottom during the chat or when I click on the chat title/file name in the history. Please advise. Thanks.

**Comment by defyzzzz** [comment]
Hello!! Very needed plugin. But I click save and I don't understand where is my chat history now?

**Comment by vanderbo** [comment]
Hello, is there a way to selectively delete chats from the history? (Could you clarify whether all chats are in the history, or only those that are "saved"?)

**Comment by defyzzzz** [comment]
It's okay, I figured it out. All chats are in place!!

**Reply by tomco582** [reply]
Can you share where you found the saved chats?

**Comment by maximromanov** [comment]
Is there a way to save some chat as a note?
